specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+of the different API objects.
+
+# Error Signalling
+
+<!-- copied from ../serving/knative-api-specification-1.0.md#error-signalling -->
+
+The Knative API uses the
+[Kubernetes Conditions convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+to communicate errors and problems to the user. Each user-visible resource
+described in Resource Overview MUST have a `conditions` field in `status`, which
+must be a list of `Condition` objects of the following form (note that the
+actual API object types may be named `FooCondition` to allow better code
+generation and disambiguation between similar fields in the same `apiGroup`):
+
+<table>
+  <tr>
+   <td><strong>Field</strong>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Default Value</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>type</code>
+   </td>
+   <td><code>string</code>
+   </td>
+   <td>The category of the condition, as a short, CamelCase word or phrase.
+<p>
+This is the primary key of the Conditions list when viewed as a map.
+   </td>
+   <td>REQUIRED – No default
+   </td>
+  </tr>
+  <tr>
+   <td><code>status</code>
+   </td>
+   <td>Enum:<ul>
+
+<li>"True"
+<li>"False"
+<li>"Unknown"</li></ul>
+
+   </td>
+   <td>The last measured status of this condition.
+   </td>
+   <td>"Unknown"
+   </td>
+  </tr>
+  <tr>
+   <td><code>reason</code>
+   </td>
+   <td>string
+   </td>
+   <td>One-word CamelCase reason for the condition's last transition.
+   </td>
+   <td>""
+   </td>
+  </tr>
+  <tr>
+   <td><code>message</code>
+   </td>
+   <td>string
+   </td>
+   <td>Human-readable sentence describing the last transition.
+   </td>
+   <td>""
+   </td>
+  </tr>
+  <tr>
+   <td><code>severity</code>
+   </td>
+   <td>Enum:<ul>
+
+<li>""
+<li>"Warning"
+<li>"Info"</li></ul>
+
+   </td>
+   <td>If present, represents the severity of the condition. An empty severity represents a severity level of "Error". 
+   </td>
+   <td>""
+   </td>
+  </tr>
+  <tr>
+   <td><code>lastTransitionTime</code>
+   </td>
+   <td>Timestamp
+   </td>
+   <td>Last update time for this condition.
+   </td>
+   <td>"" – may be unset
+   </td>
+  </tr>
+</table>
+
+Additionally, the resource's `status.conditions` field MUST be managed as
+follows to enable clients (particularly user interfaces) to present useful
+diagnostic and error message to the user. In the following section, conditions
+are referred to by their `type` (aka the string value of the `type` field on the
+Condition).
+
+1.  Each resource MUST have either a `Ready` condition (for ongoing systems) or
+    `Succeeded` condition (for resources that run to completion) with
+    `severity=""`, which MUST use the `True`, `False`, and `Unknown` status
+    values as follows:
+
+    1.  `False` MUST indicate a failure condition.
```

Comment:
agreed - so let's open a PR against serving once we're done with the edits here

Reply:
Another option is to move this stuff into a separate doc since it's common between the two and we should avoid duplicating/syncing issues.  Then future docs can just reference it too.

Reply:
That would be my preference. Do you want that in this PR, or in a separate PR?

Reply:
Changing it in this PR, per discussion at the TM meeting.